clang/include/clang/Basic/Attr.td

```diff
@@ -1313,6 +1313,40 @@ def SYCLIntelNoGlobalWorkOffset : InheritableAttr {
   let PragmaAttributeSupport = 0;
 }
 
+def SYCLIntelLoopFuse : InheritableAttr {
+  let Spellings = [CXX11<"intel","loop_fuse">];
```

Comment:
It depends on the purpose of these attributes. If these attributes are only for SYCL, then it doesn't make sense to enable them in C code, since SYCL is C++-only programming model. AFAIK we even force C++ compilation in the driver if SYCL options are used. However, we would add support for C code as well, if support for these attributes was requested for OpenCL.

Reply:
Thanks, that gets to the crux of my question, I guess: is there anything SYCL specific about loop fusion? Naively, I would have assumed this is something all backends would have some support for, but I wasn't sure.

Reply:
At the moment, loop fusion (using this attribute) is only planned for SYCL. I'll open a new issue to look into enabling this for OpenCL if required.

Reply:
Github issue to investigate C code requirements - https://github.com/intel/llvm/issues/2912